### PR TITLE
SRCH-4121 Dockerize ASIS development

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Add a Dockerfile and the configuration to the ASIS.
- ASIS service will be created and run using docker-compose.yml in search-services. Refer to https://github.com/GSA/search-services/pull/10
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Comment by ddzz** [comment]
@krbhavith ASIS checks if Elasticsearch is running when it starts up. You may need additional modifications to the code or add an `.env` file to set the `ES_HOSTS` environment variable to point to Elasticsearch.

**Comment by krbhavith** [comment]
@ddzz updated Redis config, please review.

**Review by ddzz — APPROVED** [review]
LGTM

**Comment by ddzz** [comment]
@krbhavith Actually, I remember seeing a bundler version mismatch when building the image. Has this been addressed? Same question for the I14y PR.

**Comment by krbhavith** [comment]
@ddzz addressed the  bundler version mismatch in latest commit for both i14y and asis

**Review by ddzz — DISMISSED** [review]

File: config/environments/test.rb [hunk 1]
```diff
@@ -38,4 +38,5 @@
 
   # Raises error for missing translations.
   # config.action_view.raise_on_missing_translations = true
+  config.hosts << "www.example.com"
```
Comment (ddzz) [inline config/environments/test.rb:41]:
```suggestion
  config.hosts << "www.example.com if ENV["DOCKER"]"
```

**Review by ddzz — APPROVED** [review]
LGTM

**Pull request merged**